<details><summary>Failed to submit graph for 3e5dc27e3b69509a3962c611141bb3799e4153c3
</summary>

Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmp5qnbyg3p/glib-feedstock.yaml --custom-metadata-file /tmp/tmp5qnbyg3p/custom_metadata.yaml --release-name glib-feedstock-32-3e5dc27e3b69509a3962c611141bb3799e4153c3 glib-feedstock/recipe', with error: 'INFO:root:Found feedstock glib-feedstock at /tmp/tmp5qnbyg3p/glib-feedstock/recipe
INFO:root:Build/Test graph from glib-feedstock
Created by sbx_user1051 on 2026-07-01 20:16:32
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmp5qnbyg3p/glib-feedstock.yaml --custom-metadata-file /tmp/tmp5qnbyg3p/custom_metadata.yaml --release-name glib-feedstock-32-3e5dc27e3b69509a3962c611141bb3799e4153c3 glib-feedstock/recipe

INFO:root:Found abs.yaml in /tmp/tmp5qnbyg3p/glib-feedstock
INFO:root:Creating graph from feedstock at: glib-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.11 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 4 unchanging vars:(466), changing vars (4): ['blas_impl', 'cuda_compiler_version', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_PKGS_DIRS = /tmp/conda_pkgs_bnyk4bqt.
INFO:pbpcli.utils.common:Unsetting env var: CONDA_PKGS_DIRS
INFO:pbpcli.utils.expand_variants:Elapsed time for api.render: 44.13 seconds
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-aarch64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.11 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 3 unchanging vars:(466), changing vars (3): ['cuda_compiler_version', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_OVERRIDE_GLIBC = 2.28.
INFO:pbpcli.utils.common:Setting env var: CONDA_PKGS_DIRS = /tmp/conda_pkgs_7w5jt8tz.
INFO:pbpcli.utils.common:Unsetting env var: CONDA_OVERRIDE_GLIBC
INFO:pbpcli.utils.common:Unsetting env var: CONDA_PKGS_DIRS
INFO:pbpcli.utils.expand_variants:Elapsed time for api.render: 34.62 seconds
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = osx-arm64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.11 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 2 unchanging vars:(469), changing vars (2): ['numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_OVERRIDE_OSX = 12.1.
INFO:pbpcli.utils.common:Setting env var: CONDA_PKGS_DIRS = /tmp/conda_pkgs_9tnv3_4m.
WARNING: Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
INFO:pbpcli.utils.common:Unsetting env var: CONDA_OVERRIDE_OSX
INFO:pbpcli.utils.common:Unsetting env var: CONDA_PKGS_DIRS
INFO:pbpcli.utils.expand_variants:Elapsed time for api.render: 36.78 seconds
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = win-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.11 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 4 unchanging vars:(471), changing vars (4): ['blas_impl', 'cuda_compiler_version', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_PKGS_DIRS = /tmp/conda_pkgs_8x7moyyd.
INFO:pbpcli.utils.common:Unsetting env var: CONDA_PKGS_DIRS
INFO:pbpcli.utils.expand_variants:Elapsed time for api.render: 39.44 seconds
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1524, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1445, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1912, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1912, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1308, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 877, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 321, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 611, in from_feedstock
    add_feedstock_tasks(
    ~~~~~~~~~~~~~~~~~~~^
        graph,
        ^^^^^^
    ...<6 lines>...
        parsed_abs_file,
        ^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 190, in add_feedstock_tasks
    build_task = BuildTask(
        public_name=task_name,
    ...<9 lines>...
        ),
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pydantic/main.py", line 263, in __init__
    validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 1 validation error for BuildTask
max_run_time
  Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='14400# the conda-build p...sabling --skip-existing', input_type=str]
    For further information visit https://errors.pydantic.dev/2.13/v/int_parsing
'
aws lambda request id: 1ecd052d-d9e6-5503-9b14-6792ae3bd8fc
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%601ecd052d-d9e6-5503-9b14-6792ae3bd8fc%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)

</details>

--- a/abs.yaml
+++ b/abs.yaml
@@ -1,4 +1,7 @@
 # 4 hours timeout for Windows build (default is 2 hours)
-task_timeout: 14400# the conda-build parameters to use for disabling --skip-existing
+task_timeout: 14400
+
+# the conda-build parameters to use for disabling --skip-existing
 build_parameters:
   - "--suppress-variables"
+  - "--skip-existing"

--- a/recipe/install.bat
+++ b/recipe/install.bat
@@ -53,10 +53,7 @@ if NOT [%PKG_NAME%] == [glib] (
 )
 
 @REM intl.lib is statically linked on Windows; do not advertise -lintl in .pc files.
-@REM For libglib/glib-tools, also drop .pc variables that reference pruned dev files.
-set "FIX_PC_ARGS="
-if NOT [%PKG_NAME%] == [glib] set "FIX_PC_ARGS=--trim-runtime"
-python %RECIPE_DIR%\scripts\fix-pkgconfig.py %LIBRARY_PREFIX% %FIX_PC_ARGS%
+python %RECIPE_DIR%\scripts\fix-pkgconfig.py %LIBRARY_PREFIX%
 if errorlevel 1 exit 1
 
 rem We don't have bash as a dependency so these shouldn't exist, but

--- a/recipe/install.bat
+++ b/recipe/install.bat
@@ -53,11 +53,11 @@ if NOT [%PKG_NAME%] == [glib] (
 )
 
 @REM intl.lib is statically linked on Windows; do not advertise -lintl in .pc files.
-if exist %LIBRARY_PREFIX%\lib\pkgconfig\*.pc (
-  for %%f in (%LIBRARY_PREFIX%\lib\pkgconfig\*.pc) do (
-    sed -i "s/-lintl//g" %%f
-  )
-)
+@REM For libglib/glib-tools, also drop .pc variables that reference pruned dev files.
+set "FIX_PC_ARGS="
+if NOT [%PKG_NAME%] == [glib] set "FIX_PC_ARGS=--trim-runtime"
+python %RECIPE_DIR%\scripts\fix-pkgconfig.py %LIBRARY_PREFIX% %FIX_PC_ARGS%
+if errorlevel 1 exit 1
 
 rem We don't have bash as a dependency so these shouldn't exist, but
 rem sometimes a system bash will be picked up and they will get installed.

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -103,8 +103,6 @@ outputs:
         - test -f ${PREFIX}/lib/pkgconfig/glib-2.0.pc  # [unix]
         - if not exist "%LIBRARY_LIB%\\pkgconfig\\glib-2.0.pc" exit 1  # [win]
         - findstr /C:"-lintl" "%LIBRARY_LIB%\\pkgconfig\\*.pc" >nul && exit /B 1 || exit /B 0  # [win]
-        - findstr /C:"includedir}/glib-2.0" "%LIBRARY_LIB%\\pkgconfig\\glib-2.0.pc" >nul && exit /B 1 || exit /B 0  # [win]
-        - if exist "%LIBRARY_LIB%\\pkgconfig\\gio-windows-2.0.pc" exit /B 1  # [win]
         - test -f ${PREFIX}/etc/conda/activate.d/libglib_activate.sh  # [not win]
         - test -f ${PREFIX}/etc/conda/deactivate.d/libglib_deactivate.sh  # [not win]
         # Introspection test

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -102,6 +102,9 @@ outputs:
         - test ! -f ${PREFIX}/lib/libglib-2.0${SHLIB_EXT}  # [not win]
         - test -f ${PREFIX}/lib/pkgconfig/glib-2.0.pc  # [unix]
         - if not exist "%LIBRARY_LIB%\\pkgconfig\\glib-2.0.pc" exit 1  # [win]
+        - findstr /C:"-lintl" "%LIBRARY_LIB%\\pkgconfig\\*.pc" >nul && exit /B 1 || exit /B 0  # [win]
+        - findstr /C:"includedir}/glib-2.0" "%LIBRARY_LIB%\\pkgconfig\\glib-2.0.pc" >nul && exit /B 1 || exit /B 0  # [win]
+        - if exist "%LIBRARY_LIB%\\pkgconfig\\gio-windows-2.0.pc" exit /B 1  # [win]
         - test -f ${PREFIX}/etc/conda/activate.d/libglib_activate.sh  # [not win]
         - test -f ${PREFIX}/etc/conda/deactivate.d/libglib_deactivate.sh  # [not win]
         # Introspection test

--- a/recipe/scripts/fix-pkgconfig.py
+++ b/recipe/scripts/fix-pkgconfig.py
@@ -1,0 +1,139 @@
+"""Post-process pkg-config files after meson install on Windows.
+
+Called from ``install.bat`` once per split-package output.  Meson emits
+``.pc`` files for the full development tree; the ``libglib`` and
+``glib-tools`` outputs then prune headers, tools, and schema data.  This
+script keeps the surviving ``.pc`` files consistent with what each output
+actually ships.
+"""
+from __future__ import annotations
+
+import argparse
+import pathlib
+import re
+import sys
+
+
+# Variable lines that reference binaries or data removed from runtime outputs.
+_RUNTIME_VAR_PREFIXES: dict[str, tuple[str, ...]] = {
+    "glib-2.0.pc": (
+        "glib_genmarshal=",
+        "gobject_query=",
+        "glib_mkenums=",
+        "glib_valgrind_suppressions=",
+    ),
+    "gio-2.0.pc": (
+        "schemasdir=",
+        "dtdsdir=",
+        "giomoduledir=",
+        "gio=",
+        "gio_querymodules=",
+        "glib_compile_schemas=",
+        "glib_compile_resources=",
+        "gdbus=",
+        "gdbus_codegen=",
+        "gresource=",
+        "gsettings=",
+    ),
+}
+
+# ``Cflags`` lines that point at headers pruned from runtime outputs.
+_RUNTIME_CFLAGS_PATTERNS: dict[str, tuple[re.Pattern[str], ...]] = {
+    "glib-2.0.pc": (
+        re.compile(r"^Cflags:.*glib-2\.0"),
+    ),
+    "gio-2.0.pc": (
+        re.compile(r"^Cflags:"),
+    ),
+}
+
+_RUNTIME_REMOVE_FILES = frozenset({"gio-windows-2.0.pc"})
+
+
+def _normalize_libs_line(line: str) -> str:
+    key, sep, value = line.partition(":")
+    if not sep or key not in ("Libs", "Libs.private"):
+        return line
+    value = " ".join(value.split())
+    newline = "\n"
+    if line.endswith("\r\n"):
+        newline = "\r\n"
+    elif not line.endswith("\n"):
+        newline = ""
+    return f"{key}: {value}{newline}"
+
+
+def strip_lintl(text: str) -> str:
+    """Remove ``-lintl`` from every ``Libs`` / ``Libs.private`` line."""
+    out: list[str] = []
+    for line in text.splitlines(keepends=True):
+        if line.startswith(("Libs:", "Libs.private:")):
+            line = line.replace("-lintl", "")
+            line = _normalize_libs_line(line)
+        out.append(line)
+    return "".join(out)
+
+
+def trim_runtime_pc(text: str, filename: str) -> str | None:
+    """Drop stale variables and ``Cflags`` from a runtime-output ``.pc`` file.
+
+    Returns ``None`` when the file should be deleted entirely.
+    """
+    if filename in _RUNTIME_REMOVE_FILES:
+        return None
+
+    drop_prefixes = _RUNTIME_VAR_PREFIXES.get(filename, ())
+    drop_cflags = _RUNTIME_CFLAGS_PATTERNS.get(filename, ())
+
+    out: list[str] = []
+    for line in text.splitlines(keepends=True):
+        stripped = line.lstrip()
+        if any(stripped.startswith(prefix) for prefix in drop_prefixes):
+            continue
+        if any(pattern.match(stripped) for pattern in drop_cflags):
+            continue
+        out.append(line)
+    return "".join(out)
+
+
+def process_pc_dir(
+    pc_dir: pathlib.Path,
+    *,
+    trim_runtime: bool,
+) -> None:
+    if not pc_dir.is_dir():
+        return
+
+    for pc_path in sorted(pc_dir.glob("*.pc")):
+        text = pc_path.read_text(encoding="utf-8")
+        text = strip_lintl(text)
+        if trim_runtime:
+            trimmed = trim_runtime_pc(text, pc_path.name)
+            if trimmed is None:
+                pc_path.unlink()
+                continue
+            text = trimmed
+        pc_path.write_text(text, encoding="utf-8")
+
+
+def main(argv: list[str] | None = None) -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "library_prefix",
+        type=pathlib.Path,
+        help="Windows LIBRARY_PREFIX (e.g. $PREFIX/Library)",
+    )
+    parser.add_argument(
+        "--trim-runtime",
+        action="store_true",
+        help="Prune dev-only .pc content for libglib/glib-tools outputs",
+    )
+    args = parser.parse_args(argv)
+
+    pc_dir = args.library_prefix / "lib" / "pkgconfig"
+    process_pc_dir(pc_dir, trim_runtime=args.trim_runtime)
+    return 0
+
+
+if __name__ == "__main__":
+    sys.exit(main())

--- a/recipe/scripts/fix-pkgconfig.py
+++ b/recipe/scripts/fix-pkgconfig.py
@@ -1,53 +1,15 @@
 """Post-process pkg-config files after meson install on Windows.
 
-Called from ``install.bat`` once per split-package output.  Meson emits
-``.pc`` files for the full development tree; the ``libglib`` and
-``glib-tools`` outputs then prune headers, tools, and schema data.  This
-script keeps the surviving ``.pc`` files consistent with what each output
-actually ships.
+Called from ``install.bat`` for every split-package output.  Meson emits
+``-lintl`` in ``Libs`` / ``Libs.private`` lines because libintl is linked
+on Windows, but intl is statically linked into GLib and must not be
+advertised to downstream consumers.
 """
 from __future__ import annotations
 
 import argparse
 import pathlib
-import re
 import sys
-
-
-# Variable lines that reference binaries or data removed from runtime outputs.
-_RUNTIME_VAR_PREFIXES: dict[str, tuple[str, ...]] = {
-    "glib-2.0.pc": (
-        "glib_genmarshal=",
-        "gobject_query=",
-        "glib_mkenums=",
-        "glib_valgrind_suppressions=",
-    ),
-    "gio-2.0.pc": (
-        "schemasdir=",
-        "dtdsdir=",
-        "giomoduledir=",
-        "gio=",
-        "gio_querymodules=",
-        "glib_compile_schemas=",
-        "glib_compile_resources=",
-        "gdbus=",
-        "gdbus_codegen=",
-        "gresource=",
-        "gsettings=",
-    ),
-}
-
-# ``Cflags`` lines that point at headers pruned from runtime outputs.
-_RUNTIME_CFLAGS_PATTERNS: dict[str, tuple[re.Pattern[str], ...]] = {
-    "glib-2.0.pc": (
-        re.compile(r"^Cflags:.*glib-2\.0"),
-    ),
-    "gio-2.0.pc": (
-        re.compile(r"^Cflags:"),
-    ),
-}
-
-_RUNTIME_REMOVE_FILES = frozenset({"gio-windows-2.0.pc"})
 
 
 def _normalize_libs_line(line: str) -> str:
@@ -74,45 +36,12 @@ def strip_lintl(text: str) -> str:
     return "".join(out)
 
 
-def trim_runtime_pc(text: str, filename: str) -> str | None:
-    """Drop stale variables and ``Cflags`` from a runtime-output ``.pc`` file.
-
-    Returns ``None`` when the file should be deleted entirely.
-    """
-    if filename in _RUNTIME_REMOVE_FILES:
-        return None
-
-    drop_prefixes = _RUNTIME_VAR_PREFIXES.get(filename, ())
-    drop_cflags = _RUNTIME_CFLAGS_PATTERNS.get(filename, ())
-
-    out: list[str] = []
-    for line in text.splitlines(keepends=True):
-        stripped = line.lstrip()
-        if any(stripped.startswith(prefix) for prefix in drop_prefixes):
-            continue
-        if any(pattern.match(stripped) for pattern in drop_cflags):
-            continue
-        out.append(line)
-    return "".join(out)
-
-
-def process_pc_dir(
-    pc_dir: pathlib.Path,
-    *,
-    trim_runtime: bool,
-) -> None:
+def process_pc_dir(pc_dir: pathlib.Path) -> None:
     if not pc_dir.is_dir():
         return
 
     for pc_path in sorted(pc_dir.glob("*.pc")):
-        text = pc_path.read_text(encoding="utf-8")
-        text = strip_lintl(text)
-        if trim_runtime:
-            trimmed = trim_runtime_pc(text, pc_path.name)
-            if trimmed is None:
-                pc_path.unlink()
-                continue
-            text = trimmed
+        text = strip_lintl(pc_path.read_text(encoding="utf-8"))
         pc_path.write_text(text, encoding="utf-8")
 
 
@@ -123,15 +52,10 @@ def main(argv: list[str] | None = None) -> int:
         type=pathlib.Path,
         help="Windows LIBRARY_PREFIX (e.g. $PREFIX/Library)",
     )
-    parser.add_argument(
-        "--trim-runtime",
-        action="store_true",
-        help="Prune dev-only .pc content for libglib/glib-tools outputs",
-    )
     args = parser.parse_args(argv)
 
     pc_dir = args.library_prefix / "lib" / "pkgconfig"
-    process_pc_dir(pc_dir, trim_runtime=args.trim_runtime)
+    process_pc_dir(pc_dir)
     return 0
 
 

